Add term: Javascript Objects - isExtensible (#4087)
Codecademy/docs: content/javascript/concepts/objects/terms/isExtensible/isExtensible.md

---
Title: '.isExtensible()'
Description: 'Returns a boolean value depending on whether a given object can have properties added to it.'
Subjects:

>   - 'Web Development'

Reviewer: ```suggestion
  - 'Web Development'
  - 'Code Foundation'
```

Author: Thanks! I made a slight adjustment to your suggestion, 'Code Foundations' with an "s" [from the list here](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md).

Tags:
  - 'Methods'
  - 'Objects'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/full-stack-engineer-career-path'
---

> The **`.isExtensible()`** static method returns `true` if the given object can have properties added to it and `false` if the object cannot have properties added to it.

Reviewer: ```suggestion
The **`.isExtensible()`** is a static method that returns `true` if the given object can have properties added to it and `false` if the object cannot have properties added to it.
```

## Syntax

```pseudo
Object.isExtensible(newObject);
```

Objects are extensible by default. An object can be made non-extensible by using `Object.freeze()`, `Object.preventExtensions`, ot `Object.seal()`. A non-extensible object can still have its properties modified.

## Example

> Checking if an object is extensible.

Reviewer: ```suggestion
Checking if an object is extensible using the `.isExtensible()` method:
```

```js
const characters = {
  ArthurDent: 'Earth',
  FordPrefect: 'Betelgeuse Five',
  ZaphodBeeblebrox: 'Betelgeuse Seven',
};

console.log(Object.isExtensible(characters));

Object.preventExtensions(characters);

console.log(Object.isExtensible(characters));
```

The output would be:

```shell
true
false
```

## Codebyte Example

> Following is a runnable example that demonstrates the `Object.isExtensible()` static method.

Reviewer: ```suggestion
Following is a runnable example that demonstrates the `Object.isExtensible()` static method:
```

---
Title: '.isExtensible()'
Description: 'Returns a boolean value depending on whether a given object can have properties added to it.'
Subjects:
  - 'Web Development'
  - 'Code Foundations'
Tags:
  - 'Methods'
  - 'Objects'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/full-stack-engineer-career-path'
---

The **`.isExtensible()`** is a static method that returns `true` if the given object can have properties added to it and `false` if the object cannot have properties added to it.

## Syntax

```pseudo
Object.isExtensible(newObject);
```

- `newObject`: It is the object to be checked whether it is extensible.

> > **Note:** Objects are extensible by default. An object can be made non-extensible by using `Object.freeze()`, `Object.preventExtensions`, ot `Object.seal()`. A non-extensible object can still have its properties modified.

Reviewer 2: ```suggestion
> **Note:** Objects are extensible by default. An object can be made non-extensible by using `Object.freeze()`, `Object.preventExtensions`, or `Object.seal()`. A non-extensible object can still have its properties modified.
```

---
Title: '.isExtensible()'
Description: 'Returns a boolean value depending on whether a given object can have properties added to it.'
Subjects:
  - 'Web Development'
  - 'Code Foundations'

> 

Reviewer 2: ```suggestion
```

Tags:
  - 'Methods'
  - 'Objects'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/full-stack-engineer-career-path'
---

The **`.isExtensible()`** is a static method that returns `true` if the given object can have properties added to it and `false` if the object cannot have properties added to it.

## Syntax

```pseudo
Object.isExtensible(newObject);
```

> 

Reviewer 2: Add a line describing the `newObject` parameter after the syntax block (Remember to give a line space between them).

> Objects are extensible by default. An object can be made non-extensible by using `Object.freeze()`, `Object.preventExtensions`, ot `Object.seal()`. A non-extensible object can still have its properties modified.

Reviewer 2: Add this as a note, take reference on how to add a note from Codecademy's [style guide](https://github.com/Codecademy/docs/blob/main/documentation/style-guide.md).

## Example

> Checking if an object is extensible using the `.isExtensible()` method:

Reviewer 2: ```suggestion
The following code will check if an object is extensible or not using the `.isExtensible()` method:
```

```js
const characters = {
  ArthurDent: 'Earth',
  FordPrefect: 'Betelgeuse Five',
  ZaphodBeeblebrox: 'Betelgeuse Seven',
};

console.log(Object.isExtensible(characters));

Object.preventExtensions(characters);

console.log(Object.isExtensible(characters));
```

> The output would be:

Reviewer 2: ```suggestion
The output for the above will be:
```
Try to be specific and sure as possible.

> 

Reviewer 2: ```suggestion
```

---
Title: '.isExtensible()'
Description: 'Returns a boolean value depending on whether a given object can have properties added to it.'
Subjects:
  - 'Web Development'
  - 'Code Foundations'
Tags:
  - 'Methods'
  - 'Objects'
CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/full-stack-engineer-career-path'
---

The **`.isExtensible()`** is a static method that returns `true` if the given object can have properties added to it and `false` if the object cannot have properties added to it.

## Syntax

```pseudo
Object.isExtensible(newObject);
```

- `newObject`: It is the object to be checked whether it is extensible.

> **Note:** Objects are extensible by default. An object can be made non-extensible by using `Object.freeze()`, `Object.preventExtensions`, ot `Object.seal()`. A non-extensible object can still have its properties modified.

## Example

The following code will check if an object is extensible or not using the `.isExtensible()` method:

```js
const characters = {
  ArthurDent: 'Earth',
  FordPrefect: 'Betelgeuse Five',
  ZaphodBeeblebrox: 'Betelgeuse Seven',
};

console.log(Object.isExtensible(characters));

> 

Reviewer 2: ```suggestion
```
